### DISCUSSION TITLE
How to render slot (modal) along with the rest of the page on hard refresh?

### DISCUSSION THREAD

**Discussion by aspirisen** [vercel/next.js#50133] [Help]
### Summary

Hello, I am trying to adopt [Parallel Routes](https://nextjs.org/docs/app/building-your-application/routing/parallel-routes) to show modal. The [example](https://github.com/vercel-labs/nextgram) shows the modal in empty page and I haven't found a solution how to show modal and the rest of the page according to current url.

For example I have `/cargo/quarantine` page that has data tables and other related information and I need to show filter modal by this url `/cargo/quarantine/filter` and it works for soft refresh, but when I do hard refresh the rest of the page is not rendered. There is no sense to show filter modal without the rest of the page and I am curious is there any way to render whole path, so on hard refresh I would see `/cargo/quarantine` page with the modal.

**Comment by akomm** [comment]
Documentation was not tested, just written. Understand how it works you see a lot of issues with the examples.

https://github.com/vercel/next.js/discussions/52951#discussioncomment-6852364

Just keep in mind:
layout.tsx is rendered all the way:

route `/foo/bar`
/layout.tsx
/foo/layout.tsx
/foo/bar/layout.tsx

But it matches and renders only **one** page.tsx that matches the above route:
/foo/bar/page.tsx
-> not the pages above (and it makes sense)

Its not there? Its 404
unless: you have a default.tsx at the same level export a component.

Parallel routes:
every slot in the layout.tsx must match a page.tsx in the route, or its 404. Same logic as above, but now you have not just children slot, but another one (e. G. modal or whatever you call it)

You can opt-out of 404 in a slot, by creating default.tsx on the same level and tell what to render in case the page.tsx itself does not directly match.

You can also re-export the page.tsx simply. 


/foo/layout.tsx -> renders {children} and {modal} slots
**/foo/layout.tsx** need to match both `children: /foo/page.tsx` and `modal: /foo/@modal/page.tsx ` at the same time, else its 404
/foo/page.tsx -> renders the list of foo; will only match at /foo, but not /foo/create or /foo/update
/foo/default.tsx -> import FooPage from "./page.tsx"; export default FooPage -> solves the problem that page.tsx dos not match at /foo/create and /foo/update

/foo/@modal/create/page.tsx -> renders the create modal; will only match at /foo/create, but not /foo
/foo/@modal/update/page.tsx -> renders the update modal; will only match at /foo/update, but not /foo
/foo/@modal/default.tsx -> renders null, so modal is not rendered, when you are not in /foo/create or /foo/update, otherwise you'd get 404

**Comment by akomm** [comment]
There is still a lot of mess with routing, refreshing, pushing within parallel routes. Its broken as hell.
